### PR TITLE
Create command async submission on the client side

### PR DESCRIPTION
**Pull request by luzhang6**
### What changes are proposed in this pull request?
Create command async submission on the client side:
Modify distributedCp and distributedLoad to submit commands instead of running job configs.
Modify AbstactDistributedCommand to support command waiting.
Add APIs on the job master client side.

Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  Add APIs to job master client: submit and getStatus.

### Does this PR introduce any user facing changes?
User will submit a distributed command instead of running all child jobs in a command. User will get a job control id back once command submission is successful. User can have option to wait for a command or not wait for a command to finish.

### CODE REVIEW

**Comment by luzhang6** [comment]
@yuzhu @jja725 You can skip reviews for the code change on the server side (need them to compile) since they are covered in [the other PR](https://github.com/Alluxio/alluxio/pull/15241).

**Comment by JySongWithZhangCe** [comment]
How about make changes in DistributedLoadUtils rather than DistributedLoad directly

**Comment by yuzhu** [comment]
@luzhang6 can we refactor this so it only includes changes in addition to #15241 ? it is very hard to review otherwise..

**Comment by yuzhu** [comment]
and also result in lots of conflicts

File: docs/en/operation/User-CLI.md [hunk 1]
```diff
@@ -813,7 +813,7 @@ $ ./bin/alluxio fs cp /hdfs/file1 /s3/
 ### distributedCp
 
 The `distributedCp` command copies a file or directory in the Alluxio file system distributed across workers
-using the job service.
+using the job service. User will get a job control ID after the command submission is successful.
```
Comment (yuzhu) [inline docs/en/operation/User-CLI.md:816]: User will only get a job control ID if it is async  mode right?

File: tests/src/test/java/alluxio/client/cli/fs/command/GetCmdStatusCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+
+/**
+ * Tests for getting cmd status {@link alluxio.cli.fs.command.GetCmdStatusCommand}.
+ */
+public class GetCmdStatusCommandTest extends AbstractFileSystemShellTest  {
```
Comment (yuzhu) [inline tests/src/test/java/alluxio/client/cli/fs/command/GetCmdStatusCommandTest.java:37]: Is there a test case where `--wait` is false??

**Review by yuzhu — CHANGES_REQUESTED** [review]

File: docs/en/operation/User-CLI.md (hunk 1 above)
Comment (luzhang6) [reply docs/en/operation/User-CLI.md:816]: Yes

File: tests/src/test/java/alluxio/client/cli/fs/command/GetCmdStatusCommandTest.java (hunk 2 above)
Comment (luzhang6) [reply tests/src/test/java/alluxio/client/cli/fs/command/GetCmdStatusCommandTest.java:37]: no, tested manually in cli, i will add one in unit test.

**Review by yuzhu — APPROVED** [review]
LGTM ,pending test

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by jja725** [comment]
Could you update the description of PR about what the command output looks like right now? @luzhang6

**Comment by beinan** [comment]
> How about make changes in DistributedLoadUtils rather than DistributedLoad directly

@yuzhu & @luzhang6 , I guess you two might miss at @JySongWithZhangCe 's comment.  As Ce reported offline, this pr might skip/break the features we've implemented in the DistributedLoadUtils.  any thoughts?

**Comment by yuzhu** [comment]
@JySongWithZhangCe Sorry we missed your comment earlier. Do you mind describing what behavior this might break? We can make changes to ensure it still works after this PR.

**Comment by JySongWithZhangCe** [comment]
> @JySongWithZhangCe Sorry we missed your comment earlier. Do you mind describing what behavior this might break? We can make changes to ensure it still works after this PR.

Ok~Thx:)

I have submitted 3 PRs that can make it easier to manage table data for Alluxio table users.

1. https://github.com/Alluxio/alluxio/pull/12741 Add data operation ability on AbstractTableCommand
2. https://github.com/Alluxio/alluxio/pull/13288 Packaging distributed load code into DistributedLoadUtils
3. https://github.com/Alluxio/alluxio/pull/14701 Add LoadTableCommand

DistributedLoadUtils is helpful because Alluxio table load command can sync improvement from filesystem distributedLoadUtils.

Whats more, alluxio table load command has become a good partner in muti-cluster data share, and there will be more features.

I think may be I could re-package the new design again in another PR.

**Comment by luzhang6** [comment]
> > @JySongWithZhangCe Sorry we missed your comment earlier. Do you mind describing what behavior this might break? We can make changes to ensure it still works after this PR.
> 
> Ok~Thx:)
> 
> I have submitted 3 PRs that can make it easier to manage table data for Alluxio table users.
> 
> 1. [Add data operation ability on AbstractTableCommand #12741](https://github.com/Alluxio/alluxio/pull/12741) Add data operation ability on AbstractTableCommand
> 2. [Fix packaging distributed load code into DistributedLoadUtils #13288](https://github.com/Alluxio/alluxio/pull/13288) Packaging distributed load code into DistributedLoadUtils
> 3. [Add LoadTableCommand #14701](https://github.com/Alluxio/alluxio/pull/14701) Add LoadTableCommand
> 
> DistributedLoadUtils is helpful because Alluxio table load command can sync improvement from filesystem distributedLoadUtils.
> 
> Whats more, alluxio table load command has become a good partner in muti-cluster data share, and there will be more features.
> 
> I think may be I could re-package the new design again in another PR.

@JySongWithZhangCe Thanks for sharing the info. Currently filesystem related operation is moved to be handled by job master side. Please feel free to refactor your new feature for loading table in another PR. We will take a look at the PR once you finish.

**Comment by JySongWithZhangCe** [comment]
> > > @JySongWithZhangCe Sorry we missed your comment earlier. Do you mind describing what behavior this might break? We can make changes to ensure it still works after this PR.
> > 
> > 
> > Ok~Thx:)
> > I have submitted 3 PRs that can make it easier to manage table data for Alluxio table users.
> > 
> > 1. [Add data operation ability on AbstractTableCommand #12741](https://github.com/Alluxio/alluxio/pull/12741) Add data operation ability on AbstractTableCommand
> > 2. [Fix packaging distributed load code into DistributedLoadUtils #13288](https://github.com/Alluxio/alluxio/pull/13288) Packaging distributed load code into DistributedLoadUtils
> > 3. [Add LoadTableCommand #14701](https://github.com/Alluxio/alluxio/pull/14701) Add LoadTableCommand
> > 
> > DistributedLoadUtils is helpful because Alluxio table load command can sync improvement from filesystem distributedLoadUtils.
> > Whats more, alluxio table load command has become a good partner in muti-cluster data share, and there will be more features.
> > I think may be I could re-package the new design again in another PR.
> 
> @JySongWithZhangCe Thanks for sharing the info. Currently filesystem related operation is moved to be handled by job master side. Please feel free to refactor your new feature for loading table in another PR. We will take a look at the PR once you finish.

Thx:)

**Pull request merged**